### DISCUSSION TITLE
AppRouter 에서 라우팅이 되지 않습니다.

### DISCUSSION THREAD

**Discussion by ghost** [fsoftwareengineer/todo-application#46] [질문과 답변] [answered]
아래에 해결책을 주신분이 계신데 아무리 변경해보아도 저는 라우팅이 되지 않고 빈화면만 나옵니다.
라우터 빼고 Copyright 만 띠워보거나 라우터 없이 Login 페이지만 호출할 경우 정상적으로 나왔습니다.
구글링으로 함수형으로 변경해서도 적용해보았지만 컴파일오류없이 빈화면 나와서 며칠째 진행이 안되고 있어 문의드립니다.

"react": "^17.0.2",
"react-dom": "^17.0.2",
"react-router-dom": "^6.2.1"

class AppRouter extends React.Component {
  render() {
    return (
      <div>
<Router>
          <div>
            <Routes>
              <Route path="/login" element={<Login />} />
              <Route path="/" element={<App />} />
            </Routes>
          </div>
          <Box mt={5}>
            <Copyright />
          </Box>
        </Router>
      </div>
    );
  }
}

![AppRouter](https://user-images.githubusercontent.com/88703270/147907905-e1db362a-6090-4a22-91ca-ad10f72533d0.jpg)

개발자도구 에서는 이렇게 오류가 발생하네요.
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

**Comment by fsoftwareengineer** [comment, marked as answer]
1. 프로젝트 디렉토리에서 `npm ls react`를 실행 한 후 react 버전이 하나인지 확인 해 주시기 바랍니다.
2. 빈화면이라는 것은 아얘 아무것도 나오지 않는 것인지 하단의 Copyright은 출력되는지 확인 해 주시기 바랍니다. Copyright이 출력되는 경우라면 3번을 확인 해 주시기 바랍니다. 
3. 빈화면이 나오는 상태에서 주소 창에 localhost:3000/login 을 치는 경우 라우팅이 되는지 확인 해 주시기 바랍니다. 라우팅이 되는 경우라면 App.js에 문제가 있어 렌더링이 안되는 것으로 추측 할 수 있습니다.

**Reply by iryairya** [reply]
답변 감사합니다. 
리액트는 하나였는데요. 
이것저것 해보다가 안되어서 nodejs 삭제 후 재설치 하니 정상작동 되네요.
무언가가 꼬였는데 제가 잘 몰랐던 것 같습니다. 
아참 오류화면은 Copyright 없는 완전 빈화면에 console 로그가 위에처럼 나왔습니다.

새해 복 많이 받으세요. 
작가님 책 도움 많이 받고 있습니다. 감사합니다.
